types.go

```diff
@@ -16,6 +16,10 @@ type Draining struct {
 	NodeOperator TLS
 }
 
+type GuestResources struct {
+	ClusterOperator TLS
```

Comment:
I don't follow. Are those certs to access cluster-operator?

Reply:
The certs are for cluster-operator to access the guest cluster. This is so it can install chart-operator and manage the chartconfig CRs.

Reply:
For me this isn't really intuitive. I propose changing convention a little bit.

How about something like

```
type ClusterOperator struct {
        APIServer TLS
}
```

This says those are certs for components accessed by cluster-operator. We want to make sure we have separate set of certs for each operator/service, so that's clearer approach IMO.

Reply:
I like the current approach of having the naming based on the activity. I agree it isn't clear at the moment. What about?

```
type ManageGuestResources struct {
	ClusterOperator TLS
}
```

Reply:
> I like the current approach

We don't have clear approach. It's mixed ATM. Check `Cluster` (used by {aws,azure,kvm}-operator) and `Monitoring` (used by prometheus-config-controller). They contain fields which name components that certificates allow to access (not components that use the certs). No guessing there. The only thing you have to know is what uses the certificates. So what I proposed is also an approach we have.

The problem is I see is we need to guess what `ClusterOperator TLS` allows to access. What if cluster-operator also needs to access etcd? Or prometheus? Will we have to think of another activity, name it add filed `ClusterOperator` and again guess what it allows to access?

Also when the activity in the operator changes, but it still need access will we rename those structs?

I strongly component based approach will be easier to maintain and reason about.

Reply:
👍 OK, that makes sense. Having the component and then the resources underneath it accesses is clearer. I'll change the naming.